src/trace_processor/importers/proto/winscope/surfaceflinger_layers_visibility_computation.cc

```diff
@@ -0,0 +1,361 @@
+  }
+  return !HasEmptyVisibleRegion(layer);
+}
+
+bool LayerContains(const LayerDecoder& layer,
+                   const LayerDecoder& other,
+                   const std::optional<geometry::Rect> crop) {
+  auto transform_type_layer = 0;
+  if (layer.has_transform()) {
+    protos::pbzero::TransformProto::Decoder transform(layer.transform());
+    transform_type_layer = transform.type();
+  }
+
+  auto transform_type_other = 0;
+  if (layer.has_transform()) {
+    protos::pbzero::TransformProto::Decoder transform(layer.transform());
+    transform_type_other = transform.type();
+  }
+  if (transform::IsInvalidRotation(transform_type_layer) ||
+      transform::IsInvalidRotation(transform_type_other)) {
+    return false;
+  }
+  auto layer_bounds = layer::GetCroppedScreenBounds(layer, crop);
+  auto other_bounds = layer::GetCroppedScreenBounds(other, crop);
+  return layer_bounds.has_value() && other_bounds.has_value() &&
+         layer_bounds->ContainsRect(other_bounds.value());
+}
+
+bool LayerOverlaps(const LayerDecoder& layer,
+                   const LayerDecoder& other,
+                   const std::optional<geometry::Rect> crop) {
+  auto layer_bounds = layer::GetCroppedScreenBounds(layer, crop);
+  auto other_bounds = layer::GetCroppedScreenBounds(other, crop);
+
+  return layer_bounds.has_value() && other_bounds.has_value() &&
+         layer_bounds->IntersectsRect(other_bounds.value());
+}
+
+bool IsOpaque(const LayerDecoder& layer) {
+  if (!layer.has_color()) {
+    return false;
+  }
+  ColorDecoder color(layer.color());
+  if (color.a() < 1) {
+    return false;
+  }
+  return layer.is_opaque();
+}
+
+bool IsColorEmpty(const LayerDecoder& layer) {
+  if (!layer.has_color()) {
+    return true;
+  }
+  if (HasZeroAlpha(layer)) {
+    return true;
+  }
+  return !HasValidRgb(ColorDecoder(layer.color()));
+}
+
+geometry::Rect GetDisplayCrop(
+    const LayerDecoder& layer,
+    const protos::pbzero::LayersSnapshotProto::Decoder& snapshot_decoder) {
+  geometry::Rect display_crop = geometry::Rect();
+  if (!layer.has_layer_stack()) {
+    return display_crop;
+  }
+  auto layer_stack = layer.layer_stack();
+  for (auto it = snapshot_decoder.displays(); it; ++it) {
+    protos::pbzero::DisplayProto::Decoder display(*it);
+    if (!display.has_layer_stack() || display.layer_stack() != layer_stack) {
+      continue;
+    }
+    if (!display.has_layer_stack_space_rect()) {
+      continue;
+    }
+    display_crop = display::MakeLayerStackSpaceRect(display);
+  }
+  return display_crop;
+}
+}  // namespace
+
+VisibilityComputation::VisibilityComputation(
+    const protos::pbzero::LayersSnapshotProto::Decoder& snapshot_decoder,
+    const std::vector<LayerDecoder>& layers_top_to_bottom,
+    const std::unordered_map<int32_t, LayerDecoder>& layers_by_id,
+    StringPool* pool)
+    : snapshot_decoder_(snapshot_decoder),
+      layers_top_to_bottom_(layers_top_to_bottom),
+      layers_by_id_(layers_by_id),
+      pool_(pool) {
```

Comment:
a) dont need base::StringView wrapping all the below
b) please use field initializers similar to what you have above for the rest of these.